lesson_19/homework19.2.py

```diff
@@ -0,0 +1,107 @@
+"""
+HW 19.2. POST/GET/DELETE.
+
+In the Python virtual environment (venv),
+install Flask using the command pip install flask.
+Create a file named app.py in a separate directory
+and copy the code for app.py provided in the initial data.
+Run the HTTP server using the command python app.py.
+The server will start at the base address http://127.0.0.1:8080.
+
+Using the documentation provided below, write code that:
+1. Uses the requests module to perform a POST request
+to upload an image to the server.
+2. Retrieves the link to this file using a GET request.
+3. Deletes the file from the server using a DELETE request.
+"""
+
+import logging
+
+import requests
+from requests.exceptions import HTTPError, Timeout
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s',
+)
+_log = logging.getLogger(__name__)
+
+
+def upload_to_server(base_url: str, file: str):
+    """Upload file to the server."""
+    try:
+        with open(file, 'rb') as fh:
+            files = {'image': (file, fh, 'image/png')}
+            upload_url = f'{base_url}/upload'
+            response = requests.post(upload_url, files=files, timeout=5)
+            response.raise_for_status()
+            _log.info(f'Server response: {response.json()}')
+            return response
+
+    except HTTPError as http_err:
+        _log.error(f'HTTP error: {http_err}')
+    except Timeout as time_err:
+        _log.error(f'Timeout error: {time_err}')
+    except FileNotFoundError as file_err:
+        _log.error(f'File not found: - {file_err}')
+    except Exception as err:
+        _log.error(f'An unexpected error occurred: {err}')
```

Comment:
Better to add helper function do_request and put all request error handling there. No sense to keep duplicate code for upload/get/delete operations 
Other error handling like FileNotFound could be placed to main block